pkg/rpcclient/nep11/divisible.go

```diff
@@ -0,0 +1,157 @@
+func (t *Divisible) BalanceOfD(owner util.Uint160, token []byte) (*big.Int, error) {
+	r := DivisibleReader{t.BaseReader}
+	return r.BalanceOfD(owner, token)
+}
+
+// TransferD is a divisible version of (*Base).Transfer, allowing to transfer a
+// part of NFT. It creates and sends a transaction that performs a `transfer`
+// method call using the given parameters and checks for this call result,
+// failing the transaction if it's not true. The returned values are transaction
+// hash, its ValidUntilBlock value and an error if any.
+func (t *Divisible) TransferD(from util.Uint160, to util.Uint160, amount *big.Int, id []byte, data interface{}) (util.Uint256, uint32, error) {
+	script, err := t.transferScript(from, to, amount, id, data)
+	if err != nil {
+		return util.Uint256{}, 0, err
+	}
+	return t.actor.SendRun(script)
+}
+
+// TransferDTransaction is a divisible version of (*Base).TransferTransaction,
+// allowing to transfer a part of NFT. It creates a transaction that performs a
+// `transfer` method call using the given parameters and checks for this call
+// result, failing the transaction if it's not true. This transaction is signed,
+// but not sent to the network, instead it's returned to the caller.
+func (t *Divisible) TransferDTransaction(from util.Uint160, to util.Uint160, amount *big.Int, id []byte, data interface{}) (*transaction.Transaction, error) {
+	script, err := t.transferScript(from, to, amount, id, data)
+	if err != nil {
+		return nil, err
+	}
+	return t.actor.MakeRun(script)
+}
+
+// TransferDUnsigned is a divisible version of (*Base).TransferUnsigned,
+// allowing to transfer a part of NFT. It creates a transaction that performs a
+// `transfer` method call using the given parameters and checks for this call
+// result, failing the transaction if it's not true. This transaction is not
+// signed and just returned to the caller.
+func (t *Divisible) TransferDUnsigned(from util.Uint160, to util.Uint160, amount *big.Int, id []byte, data interface{}) (*transaction.Transaction, error) {
+	script, err := t.transferScript(from, to, amount, id, data)
+	if err != nil {
+		return nil, err
+	}
+	return t.actor.MakeUnsignedRun(script, nil)
+}
+
+// Next returns the next set of elements from the iterator (up to num of them).
+// It can return less than num elements in case iterator doesn't have that many
+// or zero elements if the iterator has no more elements or the session is
+// expired.
+func (v *OwnerIterator) Next(num int) ([]util.Uint160, error) {
+	items, err := v.client.TraverseIterator(v.session, &v.iterator, num)
+	if err != nil {
+		return nil, err
+	}
+	res := make([]util.Uint160, len(items))
+	for i := range items {
+		b, err := items[i].TryBytes()
+		if err != nil {
+			return nil, fmt.Errorf("element %d is not a byte string: %w", i, err)
+		}
+		u, err := util.Uint160DecodeBytesBE(b)
+		if err != nil {
+			return nil, fmt.Errorf("element %d is not a uint160: %w", i, err)
+		}
+		res[i] = u
+	}
+	return res, nil
```

Comment:
Maybe move a part of `unwrap.ArrayOfUint160` to a separate function and reuse it here?

Reply:
Yeah, I see there is some repetitiveness in this code, but I'm not yet sure what's the best way to deal with it. There is a `stack item->proper type` pattern along with `array of items -> array of types` and iterators that are very common and probably at least some of them can be shared among different wrappers. I want to accumulate some of this code and then try to see how it can be improved.